src/Microsoft.AspNet.Razor/Parser/HtmlMarkupParser.Block.cs

```diff
@@ -558,19 +558,37 @@ private void AttributeValue(HtmlSymbolType quote)
 
             if (At(HtmlSymbolType.Transition))
             {
-                var valueStart = CurrentLocation;
-                PutCurrentBack();
+                if (NextIs(HtmlSymbolType.Transition))
+                {
+                    // Render a single "@" in place of "@@".
+                    using (Context.StartBlock(BlockType.Markup))
+                    {
+                        Span.CodeGenerator = new LiteralAttributeCodeGenerator(
+                            prefix.GetContent(prefixStart),
+                            new LocationTagged<string>(CurrentSymbol.GetContent(), CurrentLocation));
+                        AcceptAndMoveNext();
+                        Output(SpanKind.Markup, AcceptedCharacters.None);
+                        Span.CodeGenerator = SpanCodeGenerator.Null;
```

Comment:
Add a blank line above this.  Accepting and ignoring the second `@` should be more clearly separate.